distribution/helm/vector-shared/templates/_image.tpl

```diff
@@ -36,3 +36,34 @@ Resolve the full image name to use.
 {{- define "libvector.image" -}}
 {{ include "libvector.imageRepository" . }}:{{ include "libvector.imageTag" . }}
 {{- end }}
+
+{{/*
+Resolve the actual imagePullSecrets to use.
+*/}}
+{{- define "libvector.imagePullSecrets" -}}
+{{- if .Values.global }}
+  {{- if .Values.global.vector }}
+    {{- if .Values.global.vector.imagePullSecrets }}
+imagePullSecrets:
+      {{- range .Values.global.vector.imagePullSecrets }}
+- name: {{ . }}
+      {{- end }}
+    {{- else }}
+imagePullSecrets:
+      {{- range .Values.imagePullSecrets }}
+- name: {{ . }}
+      {{- end }}
+    {{- end -}}
+  {{- else }}
+imagePullSecrets:
+    {{- range .Values.imagePullSecrets }}
+- name: {{ . }}
+    {{- end }}
+  {{- end -}}
+{{- else }}
+imagePullSecrets:
+  {{- range .Values.imagePullSecrets }}
+- name: {{ . }}
+  {{- end }}
```

Comment:
What's the reason for this part to be duplicated 3 times?

Reply:
I think the ones you're referring to are for the `range` functions